src/classes/GE_GiftEntryController.cls

```diff
@@ -60,73 +64,84 @@ public with sharing class GE_GiftEntryController {
     */
     @AuraEnabled
     public static String sendPurchaseRequest(String requestBodyParameters, String dataImportRecordId) {
+        UTIL_Http.Response response = processPurchaseRequest(requestBodyParameters, dataImportRecordId);
+
+        return JSON.serialize(response);
+    }
+
+    /***
+    * @description Sends the purchase request to Elevate and logs an error on a timeout error.
+    * This method is used in tests instead of the sendPurchaseRequest() in order to skip JSON.deserialize(response)
+    * due to it erroring out because the response header contains the Object data type.
+    * @param requestBodyParameters: JSON containing parameters for the purchase call request body
+    * @param dataImportRecordId: Record ID of the related Data Import
+    * @return UTIL_Http.Response
+    */
+    @TestVisible
+    private static UTIL_Http.Response processPurchaseRequest(String requestBodyParameters, String dataImportRecordId) {
+        UTIL_Http.Response response = new UTIL_Http.Response();
```

Comment:
@kenlewis-sfdo , Great points on if and how the response should be built (response wrapper vs HttpResponse) and where it should be checked (Apex class vs JS). Feel free to change the response creation/checks/return for the purchase request, however note that create and cancel commitment functionality is using common `PS_Request` and `UTIL_Http` classes.

Reply:
Sounds good!  I think this is fine to leave as is for now and I will add my suggestions to our Gift Entry refactor list in Quip.